src/librustdoc/passes/html_tags.rs

```diff
@@ -196,14 +196,17 @@ impl<'a, 'tcx> DocFolder for InvalidHtmlTagsLinter<'a, 'tcx> {
 
             let mut tags = Vec::new();
             let mut is_in_comment = None;
+            let mut in_code_block = false;
 
             let p = Parser::new_ext(&dox, opts()).into_offset_iter();
 
             for (event, range) in p {
                 match event {
-                    Event::Html(text) | Event::Text(text) => {
+                    Event::Start(Tag::CodeBlock(_)) => in_code_block = true,
```

Comment:
Can code blocks be nested? I'm imagining something like

````markdown
# Markdown syntax

```rust
// nested code block
uiapp.run(&env::args().collect::<Vec<_>>());
```
````

Reply:
It's not supported by pulldown-cmark as it seems but I added the test just in case.